### PR TITLE
Ndbex 96802 post lh failure emails

### PR DESCRIPTION
**Pull request by evansmith**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *This work is behind a feature toggle (flipper): Partially*
-Added failure email calls to the benefits intake status check. Refactored failure email calls into their own methods to be called other places in the app.
- *(What is the solution, why is this the solution?)*
- VA Benefits and Claims
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

[- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*](https://github.com/department-of-veterans-affairs/va.gov-team/issues/96802#event-15226529803)
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: app/models/saved_claim/dependency_claim.rb [hunk 1]
```diff
@@ -134,6 +134,29 @@ def to_pdf(form_id: FORM)
     PdfFill::Filler.fill_form(self, nil, { created_at: })
   end
 
+  def send_failure_email(encrypted_user_struct = nil)
```
Comment (mjknight50) [inline app/models/saved_claim/dependency_claim.rb:137]: Can you please add a docblock here and some comments that reflect that this isn't the ideal way we wanted to go about it.  
Basically, make it clear to anyone following on behind that if we had more time, we would have done this using the common lib:
https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/va_notify/notification_email.rb

File: app/models/saved_claim/education_career_counseling_claim.rb [hunk 2]
```diff
@@ -33,4 +33,19 @@ def process_attachments!
   def business_line
     'EDU'
   end
+
+  def send_failure_email
```
Comment (mjknight50) [inline app/models/saved_claim/education_career_counseling_claim.rb:37]: Same as above, please
add docblock and some reference to how you might approach this in a refactor if you had time.

File: app/models/saved_claim/veteran_readiness_employment_claim.rb [hunk 3]
```diff
@@ -276,6 +276,22 @@ def business_line
     'VRE'
   end
 
+  def send_failure_email(encrypted_user = nil)
```
Comment (mjknight50) [inline app/models/saved_claim/veteran_readiness_employment_claim.rb:279]: Same as above, please
add docblock and some reference to how you might approach this in a refactor if you had time.

File: app/sidekiq/benefits_intake_status_job.rb [hunk 4]
```diff
@@ -168,6 +171,36 @@ def monitor_failure(form_id, saved_claim_id, bi_uuid)
         Pensions::Monitor.new.log_silent_failure(context, nil, call_location:)
       end
     end
+
+    if %w[686C-674].include?(form_id)
```
Comment (mjknight50) [inline app/sidekiq/benefits_intake_status_job.rb:175]: Would you please add the form common name for easier reading as a comment?

File: app/sidekiq/benefits_intake_status_job.rb [hunk 5]
```diff
@@ -168,6 +171,36 @@ def monitor_failure(form_id, saved_claim_id, bi_uuid)
...
+      claim = SavedClaim::DependencyClaim.find(saved_claim_id)
+      if claim
+        claim.send_failure_email
+        Dependents::Monitor.new.log_silent_failure_avoided(context, nil, call_location:)
+      else
+        Dependents::Monitor.new.log_silent_failure(context, nil, call_location:)
+      end
+    end
+
+    if %w[28-8832].include?(form_id)
```
Comment (mjknight50) [inline app/sidekiq/benefits_intake_status_job.rb:185]: Would you please add the form common name for easier reading as a comment?

File: app/sidekiq/benefits_intake_status_job.rb [hunk 6]
```diff
@@ -168,6 +171,36 @@ def monitor_failure(form_id, saved_claim_id, bi_uuid)
...
+      claim = SavedClaim::EducationCareerCounselingClaim.find(saved_claim_id)
+      if claim
+        claim.send_failure_email
+        PCPG::Monitor.new.log_silent_failure_avoided(context, nil, call_location:)
+      else
+        PCPG::Monitor.new.log_silent_failure(ocntext, nil, call_location:)
+      end
+    end
+
+    if %w[28-1900].include?(form_id)
```
Comment (mjknight50) [inline app/sidekiq/benefits_intake_status_job.rb:195]: Would you please add the form common name for easier reading as a comment?

File: app/models/saved_claim/dependency_claim.rb (hunk 1 above)
Comment (evansmith) [reply app/models/saved_claim/dependency_claim.rb:142]: done!

File: app/models/saved_claim/education_career_counseling_claim.rb (hunk 2 above)
Comment (evansmith) [reply app/models/saved_claim/education_career_counseling_claim.rb:41]: done!

File: app/models/saved_claim/veteran_readiness_employment_claim.rb (hunk 3 above)
Comment (evansmith) [reply app/models/saved_claim/veteran_readiness_employment_claim.rb:283]: done!

File: app/sidekiq/benefits_intake_status_job.rb (hunk 4 above)
Comment (evansmith) [reply app/sidekiq/benefits_intake_status_job.rb:176]: done!

File: app/sidekiq/benefits_intake_status_job.rb (hunk 5 above)
Comment (evansmith) [reply app/sidekiq/benefits_intake_status_job.rb:187]: done!

**Review by mjknight50 — APPROVED** [review]
GTG

**Comment by rmtolmach** [comment]
@evansmith it looks like spec files exist for all of the models and sidekiq jobs you added to. Can you update the tests?

**Comment by evansmith** [comment]
@rmtolmach the tests are working functionally the same. the big change for the sidekiq jobs was moving the email call inside a new method and the tests still verify that the emails go out.

**Review by micahaspyr — APPROVED** [review]
LGTM

**Comment by rmtolmach** [comment]
> @rmtolmach the tests are working functionally the same. the big change for the sidekiq jobs was moving the email call inside a new method and the tests still verify that the emails go out.

ok, I see. I guess that's fine.

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**